packages/format-library/src/text-color/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { useCallback, useMemo, useState } from '@wordpress/element';
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+import { Dashicon } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { default as InlineColorUI, getActiveColor } from './inline';
+
+const name = 'core/text-color';
+const title = __( 'Text Color' );
+
+const EMPTY_ARRAY = [];
+
+function TextColorEdit( {
+	value,
+	onChange,
+	isActive,
+	activeAttributes,
+} ) {
+	const colors = useSelect(
+		( select ) => {
+			const { getSettings } = select( 'core/block-editor' );
+			return get( getSettings(), [ 'colors' ], EMPTY_ARRAY );
+		}
+	);
+	const [ isAddingColor, setIsAddingColor ] = useState( false );
+	const enableIsAddingColor = useCallback(
+		() => ( setIsAddingColor( true ) ),
+		[ setIsAddingColor ]
+	);
+	const disableIsAddingColor = useCallback(
+		() => ( setIsAddingColor( false ) ),
+		[ setIsAddingColor ]
+	);
+	const colorIndicatorStyle = useMemo(
+		() => {
+			const activeColor = getActiveColor( name, value, colors );
+			if ( ! activeColor ) {
+				return undefined;
+			}
+			return {
+				backgroundColor: activeColor,
+			};
+		},
+		[ value, colors ]
+	);
+	return (
+		<>
+			<RichTextToolbarButton
+				className="format-library-text-color-button"
+				name={ isActive ? 'text-color' : undefined }
```

Comment:
@jorgefilipecosta - what's even more surprising is that the `name` and `key` discussed [here]('RichText.ToolbarControls') is going to be `RichText.ToolbarControls.undefined` from time to time. It all needs to be further investigated.

Reply:
There's no `Slot` for `RichText.ToolbarControls.undefined` but we should just not render this component instead.

Reply:
We need this logic because we want to show the format on the main toolbar when there is a color set, but we want to hide it from the main toolbar when color is not set.

Reply:
@epiqueras, @gziolo I think we may have a bug in slot fill, this case basically passes a dynamic name to the fill and maybe slot fill when the name changes does not remove previous fills. Setting a key seems to be a workaround for the issue.

Reply:
>So here when we pass undefined the key is going to be RichText.ToolbarControl.

So it will be in the dropdown.

>I think we may have a bug in slot fill, this case basically passes a dynamic name to the fill and maybe slot fill when the name changes does not remove previous fills. Setting a key seems to be a workaround for the issue.

Yes, see https://github.com/WordPress/gutenberg/pull/16014#discussion_r374926784.

Reply:
We should address the bug in slot&fill. But I think using a key is an ok workaround so we can merge this PR in WordPress 5.4. A key in this situation does not seem to cause problems. Any thoughts on this @gziolo, @epiqueras?

Reply:
Yes, but I’d rather have the key in the button itself to avoid this pattern from masking the need for a proper fix.